src/sqlfluff/core/rules/base.py

```diff
@@ -484,149 +567,107 @@ def _eval(self, context: RuleContext) -> EvalResultType:
 
     def crawl(
```

Comment:
Simplified the parameter list of this function. Also it is no longer recursive, now that `CrawlBehavior` does the actual iterating over the parse tree.